utils/src/main/kotlin/storage/HttpFileStorage.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.OkHttpClientHelper
+import com.here.ort.utils.getUserOrtDirectory
+
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.util.concurrent.TimeUnit
+
+import okhttp3.CacheControl
+import okhttp3.Request
+
+/**
+ * A [FileStorage] that stores files on an HTTP server.
+ */
+class HttpFileStorage(
+    /**
+     * The URL to store files at.
+     */
+    val url: String,
+
+    /**
+     * Custom headers that are added to all HTTP requests.
+     */
+    val headers: Map<String, String> = emptyMap(),
+
+    /**
+     * A path inside the [ORT user directory][getUserOrtDirectory] used for caching HTTP responses.
+     */
+    val cachePath: String? = null,
+
+    /**
+     * The max age of an HTTP cache entry in seconds.
+     */
+    val cacheMaxAgeInSeconds: Int = 0
+) : FileStorage {
+    override fun read(path: String): InputStream {
+        val builder = Request.Builder()
+            .cacheControl(CacheControl.Builder().maxAge(cacheMaxAgeInSeconds, TimeUnit.SECONDS).build())
+            .get()
+            .url(pathUrl(path))
+
+        headers.forEach { (key, value) ->
+            builder.header(key, value)
+        }
+
+        val request = builder.build()
+
+        val response = OkHttpClientHelper.execute(cachePath, request)
```

Comment:
How about `OkHttpClientHelper.execute(cachePath, request).use { response ->` here to avoid the need to explicitly close the response?

Reply:
I cannot use `use` because this would close the returned input stream before the caller can consume it.